spec/std/file_spec.cr

```diff
@@ -83,27 +84,30 @@ describe "File" do
     idx.should eq(20)
   end
 
-  it "reads lines from file with each as iterator" do
-    idx = 0
-    File.each_line("#{__DIR__}/data/test_file.txt").each do |line|
-      if idx == 0
-        line.should eq("Hello World")
+  {% unless flag?(:win32) %}
+    # TODO: these are broken on win32 because they leak file descriptors which break later tests
```

Comment:
This is a huge problem: a file can only be opened once on windows. This test tests the `each_line` iterator, which leaks a filedescriptor and therefore breaks all the tests after it.

I think the best way to solve this is to remove the `File.each_line` iterator (keep the block version) and force people to use

```cr
File.open(...) do |file|
  file.each_line...
end
```

to make leaking the file descriptor harder.

Reply:
Wouldn't be better to raise on windows inside `File.each_line` iterator, leaving this functionality on other OSes intact?

Reply:
No, because leaking file descriptors is still a problem on unix, it's just far more of a visible problem on windows.

Reply:
Another suitable alternative to `File.each_line(path)` is `File.read_lines(path).each` which reads the lines into memory and iterates over the array.